op-supervisor/supervisor/backend/db/query.go

```diff
@@ -10,6 +10,31 @@ import (
 )
 
 func (db *ChainsDB) FindSealedBlock(chain eth.ChainID, number uint64) (seal types.BlockSeal, err error) {
+	// Check if in pre-activation mode
+	if db.IsInPreActivationMode(chain) {
+		status, ok := db.GetPreActivationStatus(chain)
+		if !ok {
+			return types.BlockSeal{}, types.ErrFuture
+		}
+
+		// Only return the head block in pre-activation mode
+		head := status.LocalUnsafe
+		if number != head.Number {
+			if number > head.Number {
+				return types.BlockSeal{}, types.ErrFuture
+			}
+			return types.BlockSeal{}, types.ErrConflict
```

Comment:
So it's guaranteed that all calls to the functions in this file, in pre-activation mode, aren't called for older blocks, not at tip?